File: 31-next-permutation/next-permutation.py
class Solution:
    def nextPermutation(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        temp = -1
        for i in range(len(nums)-2, -1, -1):
            if nums[i] < nums[i+1] : 
                temp = i
                break
        if temp == -1:
            return nums.reverse()
        for i in range(len(nums)-1, temp, -1):
            if nums[temp] < nums[i]:
                nums[temp], nums[i] = nums[i], nums[temp]
                break
        nums[temp+1:] = reversed(nums[temp+1:])

File: 31-next-permutation/next-permutation.cpp
class Solution {
public:
    void nextPermutation(vector<int>& nums) {
        int temp = -1;
        for(int i=nums.size()-2;i>=0;i--){
            if(nums[i] < nums[i + 1]){
                temp = i;
                break;
            }
        }
        if(temp == -1){
            reverse(nums.begin(), nums.end());
            return;
        }
        for(int i=nums.size()-1;i>=temp;i--){
            if(nums[temp] < nums[i]){
                swap(nums[temp], nums[i]);
                break;
            }
        }
        reverse(nums.begin()+temp+1, nums.end());
    }
};